### PR TITLE
Handle null/empty type paths in build url mixin

### PR DESCRIPTION
**Pull request by vinilios**
handle empty values returned by pathForType in buildURL method of
the build url mixin.

refs emberjs/data#2844

### CODE REVIEW

File: packages/ember-data/lib/adapters/build-url-mixin.js [hunk 1]
```diff
@@ -52,8 +52,11 @@ export default Ember.Mixin.create({
     var url = [];
     var host = get(this, 'host');
     var prefix = this.urlPrefix();
+    var path;
 
-    if (type) { url.push(this.pathForType(type)); }
+    if (type && (path = this.pathForType(type))) {
```
Comment (igorT) [inline packages/ember-data/lib/adapters/build-url-mixin.js]: Can you take the assignment out of the if statement? I don't think it's needed there.

**Comment by igorT** [comment]
Thanks for the PR!

**Comment by vinilios** [comment]
PR updated @igorT

**Comment by igorT** [comment]
Thanks!

**Pull request merged**